AllReadyApp/Web-App/AllReady/Features/Events/UnregisterEventHandler.cs

```diff
@@ -1,27 +1,42 @@
 ﻿using System.Threading.Tasks;
+using System.Linq;
 using AllReady.Features.Notifications;
 using AllReady.Models;
 using MediatR;
 
 namespace AllReady.Features.Event
 {
-    public class UnregisterEventHandler : AsyncRequestHandler<UnregisterEvent>
-    {
-        private readonly IAllReadyDataAccess dataAccess;
+    public class UnregisterEventHandler : AsyncRequestHandler<UnregisterEvent> {
+        private readonly AllReadyContext dataContext;
         private readonly IMediator mediator;
 
-        public UnregisterEventHandler(IAllReadyDataAccess dataAccess, IMediator mediator)
-        {
-            this.dataAccess = dataAccess;
+        public UnregisterEventHandler(AllReadyContext dataContext, IMediator mediator) {
+            this.dataContext = dataContext;
             this.mediator = mediator;
         }
 
-        protected override async Task HandleCore(UnregisterEvent message)
-        {
-            await dataAccess.DeleteEventAndTaskSignupsAsync(message.EventSignupId).ConfigureAwait(false);
-            
+        protected override async Task HandleCore(UnregisterEvent message) {
+            await this.DeleteEventAndTaskSignupsAsync(message.EventSignupId).ConfigureAwait(false);
+
             //Notify admins & volunteer
-            await mediator.PublishAsync(new UserUnenrolls { EventId = message.EventSignupId, UserId = message.UserId }).ConfigureAwait(false);
+            await mediator.PublishAsync(new UserUnenrolls {EventId = message.EventSignupId, UserId = message.UserId}).ConfigureAwait(false);
+        }
+
+        // public to be testable, not meant for public consumption
+        public Task DeleteEventAndTaskSignupsAsync(int eventSignupId) {
+            var eventSignup = this.dataContext.EventSignup.SingleOrDefault(c => c.Id == eventSignupId);
+
+            if (eventSignup == null) {
+                return Task.FromResult(0);
+            }
+
+            this.dataContext.EventSignup.Remove(eventSignup);
+
+            this.dataContext.TaskSignups.RemoveRange(this.dataContext.TaskSignups
```

Comment:
That's cool our preferences differ here.  I prefer `this` as it highlights instantly this isn't a local variable.  I also like to avoid `var` as `int` and `string` are much more descriptive.  I tried to leave the original author's style unchanged where possible, though R# likes to 'help'.